__tests__/Login.test.js

```diff
@@ -34,22 +43,24 @@ describe('Login Component', () => {
     expect(wrapper.find(LoginForm)).toHaveLength(1);
   });
 
-  it('should render its children', () => {
-    expect(wrapper.find('div').length).toBe(7);
-    expect(wrapper.find('form').length).toBe(1);
-    expect(wrapper.find('h4').length).toBe(1);
-    expect(wrapper.find('label').length).toBe(2);
-    expect(wrapper.find('input').length).toBe(2);
-    expect(wrapper.find('button').length).toBe(1);
-
-  })
-
-
-  it('should call loginUser action', () => {
+  it('should call handleSubmit when login form is submitted', () => {
+    const spy = jest.spyOn(wrapper.instance(), 'handleSubmit');
     const form = wrapper.find('form');
-    const hadleSubmit = jest.spyOn(wrapper.instance(), 'handleSubmit');
-    wrapper.instance().handleSubmit(event);
-    expect(hadleSubmit).toBeCalled();
+    wrapper.instance().forceUpdate();
+    form.simulate('submit');
+    expect(spy).toBeCalled();
   });
 
-});
\ No newline at end of file
+  it('should call handleChange on login form input change', () => {
+    const spy = jest.spyOn(wrapper.instance(), 'handleChange');
+    const form = wrapper.find('form');
+    wrapper.instance().forceUpdate()
+    form.find("input[name='username']").simulate('change', {
+      target: {
+        value: "testUsername"
+      }
+    });
+    expect(spy).toBeCalled();
+  })
+
+});
```

Comment:
Hi @dondrzzy 
How about you test if the `handleChange` method is actually updating the state with the inputs since that is its main purpose?

Reply:
@IAMOTZ, that's very wise. I have added a test that actually tests what the function does.